back/src/domains/establishment/adapters/PgEstablishmentAggregateRepository.ts

```diff
@@ -667,31 +660,34 @@ export class PgEstablishmentAggregateRepository
     inseeCheckDate: Date,
     params: UpdateEstablishmentsWithInseeDataParams,
   ): Promise<void> {
-    const queries = keys(params).map((siret) => {
-      const values = params[siret];
-      return format(
-        `
-            UPDATE establishments
-            SET last_insee_check_date = %1$L 
-              ${values?.isOpen !== undefined ? ", is_open=%3$L" : ""}
-              ${values?.nafDto ? ", naf_code=%4$L" : ""}
-              ${values?.nafDto ? ", naf_nomenclature=%5$L" : ""}
-              ${values?.name ? ", name=%6$L" : ""}
-              ${values?.numberEmployeesRange ? ", number_employees=%7$L" : ""}
-            WHERE siret = %2$L;`,
-        inseeCheckDate.toISOString(),
-        siret,
-        ...[
-          values?.isOpen,
-          values?.nafDto?.code,
-          values?.nafDto?.nomenclature,
-          values?.name,
-          values?.numberEmployeesRange,
-        ],
-      );
-    });
-
-    await executeKyselyRawSqlQuery(this.transaction, queries.join("\n"));
+    await Promise.all(
+      toPairs(params).map(([siret, values]) => {
+        return this.transaction
+          .updateTable("establishments")
+          .set({
+            last_insee_check_date: inseeCheckDate,
+            ...(values !== undefined
+              ? {
+                  ...(values.isOpen !== undefined
+                    ? { is_open: values.isOpen }
+                    : {}),
+                  ...(values.nafDto !== undefined
+                    ? {
+                        naf_code: values.nafDto.code,
+                        naf_nomenclature: values.nafDto.nomenclature,
+                      }
+                    : {}),
+                  ...(values.name !== undefined ? { name: values.name } : {}),
+                  ...(values.numberEmployeesRange !== undefined
+                    ? { number_employees: values.numberEmployeesRange }
+                    : {}),
+                }
+              : {}),
+          })
+          .where("siret", "=", siret)
+          .execute();
+      }),
+    );
```

Comment:
Ici on passe d'un appel avec plusieurs requêtes à plusieurs appels mais d'une requête à la fois. Je ne sais pas si on peut les fusionner en un seul appel et une seule requête et si un seul appel avec plusieurs requêtes apporte un gain niveau perf.

Reply:
ce qui me fait un peu peur c'est le promise.all, potentiellement tu envoie des centaines de connection à la DB en même temps. Il y a un petit utilitaire pour faire les appels en séquence `executeInSequence` , peut-être que c'est mieux. Ça ferait les appels 1 par 1. Ce qui pour le coups est peut-être un peu faible...

Reply:
je peux faire une boucle for à la place